host/pages/index.tsx

```diff
@@ -34,7 +39,9 @@ export default function Home() {
         <PathwayViewerComponent />
         <br></br>
         <hr />
-        <ExternalComponentLoader extensionName={extensionName} />
+        <ApolloProvider client={starshipsApolloClient} >
```

Comment:
This is where I added the additional ApolloProvider. Now I am still not sure if it was allowed for me to modify the hosted page - I am assuming the developer has control of both the hosted page and extensions.  It is possible to create a client at the extension or even at the hook level I think so that would be an option if this is not acceptable. Also, I so some libraries which allow for multiple ApolloClients to switch so if we wanted to support both thats an option as well.

Reply:
i think it would be a better assumption that an extension developer doesn't need to access hosted pages at all (if you recall, one of the two problems we were trying to solve was the coupling between hosted pages and extensions)

Reply:
True. That makes sense now that you say it. I think lost track of the distinctions between hosted page and extensions - thinking that hosted pages are actually an accessory that comes with an extension. Like a form that is needed to complete the careflow. And since the other networking (other Apollo provider) was at the hosted page level, I tried to keep it all there. But yeah I started to question that assumption so thats why i wrote it down here.

Reply:
Also the other thought I had, was that starships service was a "service" that would be available to all the extensions of the hosted page. And providing at  the hosted level provides it to all the extensions. So I think I assume that we we wanted it to be coupled. Just a misunderstanding on my part how extensions and hosted pages are modeled /intended for the developers.

Reply:
Fixed it - added Apollo provider at extension level. Also refactored the "Starship" code to be in Starship component to allow it to pass the apollo provider in nice config like syntax.